src/component/mxgraph/shape/text-annotation-shapes.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2020 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { mxgraph } from 'ts-mxgraph';
+import { StyleDefault } from '../StyleUtils';
+
+export class TextAnnotationShape extends mxRectangleShape {
```

Comment:
The **mxText** is not more adapted than **mxRectangleShape** ?

Reply:
in fact text itself is handled by Label mechanism
here only the outer rectangular shape is needed